src/backend/joanie/core/models/contracts.py

```diff
@@ -218,10 +219,32 @@ class Meta:
             ),
         ]
 
+    # pylint: disable=no-member
     def __str__(self):
         return f"{self.order.owner}'s contract for course {self.order.course}"
 
     def save(self, *args, **kwargs):
         """Enforce validation each time an instance is saved."""
         self.full_clean()
         super().save(*args, **kwargs)
+
+    def tag_submission(self, reference, checksum, context):
```

Comment:
actually submission is not explicit enough
```suggestion
    def tag_submission_for_signature(self, reference, checksum, context):
```

Reply:
It is better, I agree with you